pathspider/cmd/measure.py

```diff
@@ -107,6 +108,17 @@ def run_measurement(args):
                 outputfile.write(json.dumps(result) + "\n")
                 logger.debug("wrote a result")
                 spider.outqueue.task_done()
+        if args.upload is not None:
```

Comment:
Upload should be a distinct step that happens after the measurement. In the arrangement here it is possible to run the entire measurement only to have it fail at the end because no filename was specified. That is not great.

Please do not add this to the measurement code but instead as a separate command only.

Reply:
So do you want all changes to the measure.py removed?

Reply:
Ideally yes.